### PR TITLE
Udemy course section 5: front end styling and bootstrap

### PR DESCRIPTION
**Pull request by ghost**
For this part of the course, I added stylings to the backend which I have built so far. We used bootstrap and erb to improve the design and useability of the app.

Please check if what I have built so far follows about source conventions.

### CODE REVIEW

**Comment by ghost** [comment]
When reviewing a PR it is very helpful to know what it is about. It is a good practice to give your PR a good title and maybe a short description as well (if necessary).

File: app/views/layouts/_messages.html.erb [hunk 1]
```diff
@@ -1,3 +1,10 @@
...
+<div class="container">
+  <% flash.each do |name, msg| %>
+    <div class="alert alert-success alert-dismissible fade show" role="alert">
+      <%= msg %>
+      <button type="button" class="close" data-dismiss="alert" aria-label="Close">
+        <span aria-hidden="true">&times;</span>
+      </button>
+    </div>
+  <% end %>
+</div>
```
Comment (ghost) [inline app/views/layouts/_messages.html.erb]: missing new line at the end

File: config/environment.rb [hunk 2]
```diff
@@ -3,3 +3,8 @@
 
 # Initialize the Rails application.
 Rails.application.initialize!
+
+# To disable extra div when you hit a validation error message
+ActionView::Base.field_error_proc = Proc.new do |html_tag, instance|
+  html_tag.html_safe
+end
```
Comment (ghost) [inline config/environment.rb]: missing newline

**Review by ghost — CHANGES_REQUESTED** [review]
ok, das ist ja alles frontend. Nicht so spannend für mich. Sieht alles gut, nur 2 Zeilenumbrüche fehlen.

File: app/assets/stylesheets/custom.css.scss [hunk 3]
```diff
@@ -0,0 +1,42 @@
+@import 'bootstrap/dist/css/bootstrap';
```
Comment (aboutsimon) [inline app/assets/stylesheets/custom.css.scss:1]: You can import the SCSS files of Bootstrap directly.

```suggestion
@import '~bootstrap';
```

see https://getbootstrap.com/docs/4.0/getting-started/theming/#importing

Comment (aboutsimon) [reply app/assets/stylesheets/custom.css.scss:1]: Rename file to `custom.scss`? Or is there any reason for the `.css.scss`?

**Review by aboutsimon — COMMENTED** [review]
Just some minor frontend suggestions :)

File: app/assets/stylesheets/custom.css.scss (hunk 3 above)
Comment (ghost) [reply app/assets/stylesheets/custom.css.scss:1]: Hey @aboutsimon cool, good question... To be honest, I have no idea. This is the code we are taught in the course. What is the difference between `scss` und `css.scss`?

Comment (aboutsimon) [reply app/assets/stylesheets/custom.css.scss:1]: Well, `.scss` is the extension of the [css preprocessor SASS](https://sass-lang.com/) (don't be confused by sass and scss, they are just two different flavors of SASS, with SCSS being the more popular one). `.css` just stand's for plain old css. The `*.css.scss` is probably just a weird Rails default?!

File: config/environment.rb (hunk 2 above)
Comment (ghost) [reply config/environment.rb]: Added

File: app/assets/stylesheets/custom.css.scss (hunk 3 above)
Comment (ghost) [reply app/assets/stylesheets/custom.css.scss:1]: OK, so should I make the changes you suggest or leave it for now? I don't mind which way :)

**Review by ghost — APPROVED** [review]
looks great, very good title and description, thank you!

File: app/assets/stylesheets/custom.css.scss (hunk 3 above)
Comment (aboutsimon) [reply app/assets/stylesheets/custom.css.scss:1]: I would leave the filename as is and try changing the import statement to `@import '~bootstrap';` as suggested above. That way, you could [theme your bootstrap instance](https://getbootstrap.com/docs/4.0/getting-started/theming/), if you want.

Comment (aboutsimon) [reply app/assets/stylesheets/custom.css.scss:1]: but don't get hung up on that, this is not a merge blocker ;)

**Comment by ghost** [comment]
Thank you for all your comments, I am closing the PR and have deleted the branch.

**Pull request merged**